src/components/panel/panel.js

```diff
@@ -1662,56 +1685,93 @@ MdPanelPosition.prototype.absolute = function() {
   return this;
 };
 
+/**
+ * Sets the value of a position for the panel. Clears any previously set position.
+ * @param {string} position Position to set
+ * @param {string=} value Value of the position. Defaults to '0'.
+ * @returns {MdPanelPosition}
+ * @private
+ */
+MdPanelPosition.prototype._setPosition = function(position, value) {
+  if (position === 'right' || position === 'left') {
+    this._left = this._right = '';
+  }
+  else if (position === 'bottom' || position === 'top') {
+    this._top = this._bottom = '';
+  }
+  else {
+    throw new Error('Position must be one of \'top\', \'bottom\', \'left\' or \'right\'.');
+  }
+
+  this['_' +  position] = value || '0';
```

Comment:
Can you assert that value is actually a string? I don't want anyone setting a function here. This line screams hacker to me.

if (!angular.isString(value)) {
  throw new Error('position value must be a string.');
}

// Written on my phone without back ticks.